sdk/storage/azure-storage-quickquery/src/test/java/com/azure/storage/quickquery/FluxInputStreamAPITest.groovy

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.quickquery
+
+import com.azure.core.http.rest.ResponseBase
+import com.azure.core.test.http.MockHttpResponse
+import com.azure.core.util.logging.ClientLogger
+import com.azure.storage.blob.models.BlobRequestConditions
+import com.azure.storage.blob.models.BlobStorageException
+import com.azure.storage.common.implementation.Constants
+import com.azure.storage.quickquery.implementation.util.FluxInputStream
+import com.azure.storage.quickquery.models.BlobQuickQuerySerialization
+import org.eclipse.jetty.util.IO
+import reactor.core.publisher.Flux
+import spock.lang.Unroll
+
+import java.nio.ByteBuffer
+
+class FluxInputStreamAPITest extends APISpec {
+
+    Flux<ByteBuffer> generateData(int num) {
+        List<ByteBuffer> buffers = new ArrayList<>()
+        for(int i = 0; i < num; i++) {
+            buffers.add(ByteBuffer.wrap(i.byteValue()))
+        }
+        return Flux.fromIterable(buffers)
+    }
+
+    @Unroll
+    def "NetworkIS min"() {
+        setup:
+        def data = generateData(num)
+
+        when:
+        def is = new FluxInputStream(data, new ClientLogger(FluxInputStream.class))
+        def bytes = new byte[num]
+
+        def totalRead = 0
+        def bytesRead = 0
+
+        while (bytesRead != -1 && totalRead < num) {
+            bytesRead = is.read(bytes, totalRead, num)
+            if (bytesRead != -1) {
+                totalRead += bytesRead
+                num -= bytesRead
+            }
+        }
+
+        is.close()
+
+        then:
+        for (int i = 0; i < num; i++) {
+            assert bytes[i] == i.byteValue()
+        }
+
```

Comment:
Can you add a couple more checks to assert that everything is also the same size? Like is.available == 0 and totalRead == num?